src/artefacts_demo_control/test/artefacts_demo_control/collision_test.py

```diff
@@ -85,44 +85,55 @@ def test_collsion(self):
             x_variable = float(match.group(1))
             y_variable = float(match.group(2))
             z_variable = float(match.group(3))
-            print(f"x: {x_variable}")
-            print(f"y: {y_variable}")
-            print(f"z: {z_variable}")
+            print(f"x box coordinate: {x_variable}")
+            print(f"y box coordinate: {y_variable}")
+            print(f"z box coordinate: {z_variable}")
         else:
             print("No x, y, or z variables found in the text.")
             x_variable = 0.0
             y_variable = 0.0
             z_variable = 0.0
+        return x_variable, y_variable, z_variable
 
-        sleep(30)
+    def test_box_moved(self, proc_output, demo):
+        """
+        Test case to see if box has moved
+        """
 
-        command = ["gz", "model", "-m", "artefacts_box", "-i"]
-        result = subprocess.run(command, check=True, text=True, stdout=subprocess.PIPE)
-        result_output = result.stdout
+        sleep(10)
 
-        # Extract new position data if it exists
-        position_pattern = r"pose\s*{\s*position\s*{\s*x:\s*(-?\d+\.\d+)\s*y:\s*(-?\d+\.\d+)\s*z:\s*(-?\d+\.\d+)"
+        (
+            TestCollision.x_variable,
+            TestCollision.y_variable,
+            TestCollision.z_variable,
+        ) = self.get_box_location()
 
-        # Search for the new x, y, and z variables using the regular expression
-        match = re.search(position_pattern, result_output, re.DOTALL)
+        proc_output.assertWaitFor("Block has been moved", timeout=180)
 
-        # Extract the new x, y, and z variables if a match is found
-        if match:
-            x_new_variable = float(match.group(1))
-            y_new_variable = float(match.group(2))
-            z_new_variable = float(match.group(3))
-            print(f"x new variable: {x_new_variable}")
-            print(f"y new variable: {y_new_variable}")
-            print(f"z new variable: {z_new_variable}")
-        else:
-            print("No x, y, or z variables found in the text.")
-            x_new_variable = 0.0
-            y_new_variable = 0.0
-            z_new_variable = 0.0
+        (
+            TestCollision.x_new_variable,
+            TestCollision.y_new_variable,
+            TestCollision.z_new_variable,
+        ) = self.get_box_location()
 
-        sleep(10)
 
+@launch_testing.post_shutdown_test()
+class TestAfterShutdown(unittest.TestCase):
+    def test_collsion(self):
+        """
+        Collision Test case, if the values are no longer equal it means that the block has shifted position through a collison
+        """
         decimal = 3
-        self.assertAlmostEqual(x_variable, x_new_variable, decimal)
-        self.assertAlmostEqual(y_variable, y_new_variable, decimal)
-        self.assertAlmostEqual(z_variable, z_new_variable, decimal)
+
+        self.assertAlmostEqual(
```

Comment:
True, but this allows for a cleaner test since it no allows for gazebo to shutdown and there no longer is the error of 0 tests ran after gazebo closes.